sfge/src/main/java/com/salesforce/graph/build/StaticBlockUtil.java

```diff
@@ -0,0 +1,345 @@
+package com.salesforce.graph.build;
+
+import com.salesforce.apex.jorje.ASTConstants;
+import com.salesforce.apex.jorje.JorjeNode;
+import com.salesforce.collections.CollectionUtil;
+import com.salesforce.exception.ProgrammingException;
+import com.salesforce.graph.Schema;
+import com.salesforce.graph.ops.MethodUtil;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeSet;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+/**
+ * Handles creation of synthetic methods and vertices to gracefully invoke static code blocks.
+ *
+ * <p>Consider this example: class StaticBlockClass { static { System.debug("inside static block
+ * 1"); } static { System.debug("inside static block 2"); } } In Jorje's compilation structure,
+ * static blocks are represented like this: class StaticBlockClass { private static void <clint>() {
+ * { System.debug("inside static block 1"); } { System.debug("inside static block 2"); } } }
+ *
+ * <p>Having multiple block statements inside a method breaks SFGE's makes handling code blocks in
+ * <clint>() impossible. As an alternative, we are creating synthetic vertices in the Graph to
+ * represent the static blocks as individual methods. We also create one top-level synthetic method
+ * ("StaticBlockInvoker") that invokes individual static block methods. While creating static scope
+ * for this class, we should invoke the method call expressions inside the top-level synthetic
+ * method.
+ *
+ * <p>New structure looks like this: class StaticBlockClass { private static void
+ * SyntheticStaticBlock_1() { System.debug("inside static block 1"); }
+ *
+ * <p>private static void SyntheticStaticBlock_2() { System.debug("inside static block 2"); }
+ *
+ * <p>private static void StaticBlockInvoker() { SyntheticStaticBlock_1(); SyntheticStaticBlock_2();
+ * } }
+ */
+public final class StaticBlockUtil {
+    private static final Logger LOGGER = LogManager.getLogger(StaticBlockUtil.class);
+
+    static final String SYNTHETIC_STATIC_BLOCK_METHOD_NAME = "SyntheticStaticBlock_%d";
+    static final String STATIC_BLOCK_INVOKER_METHOD = "StaticBlockInvoker";
+
+    private StaticBlockUtil() {}
+
+    /**
+     * Creates a synthetic method vertex to represent a static code block
+     *
+     * @param g traversal graph
+     * @param clintVertex <clint>() method's vertex
```

Comment:
It's `clinit`, not `clint`. It's a little thing, but it's in a bunch of places and it'll bug me if I don't mention it.

Reply:
Thanks for the catch, I'll fix it.

Reply:
Fixed everywhere